src/twister2/builder/build_helper.py

```diff
@@ -0,0 +1,168 @@
+            build_config.zephyr_base,
+            build_config.build_dir,
+            build_config.platform_arch,
+            build_config.platform_name,
+            build_config.cmake_filter
+        )
+        result: bool = cmake_filter.filter()
+        if result is False:
+            msg = 'runtime filter - build configuration do not fullfil Kconfig, DTS or other CMake requirements'
+            logger.info(msg)
+            raise TwisterBuildFiltrationException(msg)
+
+
+@dataclass
+class CMakeExtraArgsConfig:
+    """Class store all information required to generate extra CMake extra args."""
+    run_id: str
+    build_dir: Path
+    device_type: str
+    fifo_file: Path
+    platform_arch: str
+    platform_name: str
+    extra_args_spec: list[str] = field(default_factory=list)
+    extra_configs: list[str] = field(default_factory=list)
+    extra_args_cli: list[str] = field(default_factory=list)
+
+
+class CMakeExtraArgsGenerator:
+    def __init__(self, config: CMakeExtraArgsConfig) -> None:
+        self.config = config
+
+    def generate(self) -> list[str]:
+
+        cmake_args = []
+        cmake_args.append(f'-DTC_RUNID={self.config.run_id}')
+        cmake_args += self._prepare_warning_as_error_args()
+
+        extra_args = self.config.extra_args_spec
+        extra_args += self._prepare_extra_configs()
+
+        cmake_args += self._prepare_extra_args_spec(extra_args)
+
+        if self.config.device_type == 'qemu':
+            cmake_args.append(f'-DQEMU_PIPE={str(self.config.fifo_file)}')
+
+        cmake_args += self._prepare_extra_args_cli(self.config.extra_args_cli)
+
+        return cmake_args
+
+    @staticmethod
+    def _prepare_warning_as_error_args() -> list[str]:
+        ldflags = '-Wl,--fatal-warnings'
+        cflags = '-Werror'
+        aflags = '-Werror -Wa,--fatal-warnings'
+        gen_defines_args = '--edtlib-Werror'
+
+        args = [
+            f'-DEXTRA_CFLAGS={cflags}',
+            f'-DEXTRA_AFLAGS={aflags}',
+            f'-DEXTRA_LDFLAGS={ldflags}',
+            f'-DEXTRA_GEN_DEFINES_ARGS={gen_defines_args}',
+        ]
+        return args
+
+    def _prepare_extra_configs(self) -> list[str]:
+        extra_configs_overlay: str = ''
+
+        if self.config.extra_configs:
+            parsed_extra_configs = self._parse_extra_configs()
+            extra_configs_overlay = '\n'.join(parsed_extra_configs)
+
+        additional_extra_args = []
+        if extra_configs_overlay:
+            extra_configs_file_path = self._export_overlay_config(self.config.build_dir, extra_configs_overlay)
+            additional_extra_args.append(f'OVERLAY_CONFIG="{str(extra_configs_file_path)}"')
+
+        return additional_extra_args
+
+    def _parse_extra_configs(self) -> list[str]:
+        """
+        Some configs might be conditional on arch or platform, see if we
+        have a namespace defined and apply only if the namespace matches.
+        we currently support both arch: and platform:
```

Comment:
Typo:
`we currently support both arch**:** and platform:`

Reply:
Fixed